lottie/src/main/java/com/airbnb/lottie/parser/GradientColorParser.java

```diff
@@ -139,7 +139,7 @@ private int getOpacityAtPosition(double position, double[] positions, double[] o
       double lastPosition = positions[i - 1];
       double thisPosition = positions[i];
       if (positions[i] >= position) {
-        double progress = (position - lastPosition) / (thisPosition - lastPosition);
+        double progress = MiscUtils.clamp((position - lastPosition) / (thisPosition - lastPosition), 0, 1);
```

Comment:
I assume it would be fine to change the precision of the progress and use an existing clamp method. I didn't want to make that change without some more knowledge about the choices here though.